pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -115,6 +128,27 @@ func deletePodHandler(c clientset.Interface, emitEventFunc func(types.Namespaced
 	}
 }
 
+func addConditionAndDeletePod(ctx context.Context, c clientset.Interface, name, ns string) (err error) {
+	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+		pod, err := c.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		newStatus := pod.Status.DeepCopy()
+		if apipod.UpdatePodCondition(newStatus, &v1.PodCondition{
+			Type:    targetedForDisruption,
+			Status:  v1.ConditionTrue,
+			Reason:  "DeletionByTaintManager",
+			Message: "To be deleted by taint manager due to the NoExecute taint",
```

Comment:
"taint manager: deleting due to NoExecute taint"

Reply:
updated, just capitalized the first letter